app/gosqs/send_message_batch.go

```diff
@@ -0,0 +1,108 @@
+package gosqs
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Admiral-Piett/goaws/app"
+	"github.com/Admiral-Piett/goaws/app/common"
+	"github.com/Admiral-Piett/goaws/app/interfaces"
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/Admiral-Piett/goaws/app/utils"
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+)
+
+func SendMessageBatchV1(req *http.Request) (int, interfaces.AbstractResponseBody) {
+
+	requestBody := models.NewSendMessageBatchRequest()
+	ok := utils.REQUEST_TRANSFORMER(requestBody, req, false)
+
+	if !ok {
+		log.Error("Invalid Request - SendMessageBatchV1")
+		return utils.CreateErrorResponseV1("InvalidParameterValue", true)
+	}
+
+	queueUrl := requestBody.QueueUrl
+
+	// TODO: Remove this query param logic if it's not still valid or something
+	queueName := ""
+	if queueUrl == "" {
+		vars := mux.Vars(req)
+		queueName = vars["queueName"]
+	} else {
+		uriSegments := strings.Split(queueUrl, "/")
+		queueName = uriSegments[len(uriSegments)-1]
+	}
+
+	if _, ok := app.SyncQueues.Queues[queueName]; !ok {
+		return utils.CreateErrorResponseV1("QueueNotFound", true)
+	}
+
+	sendEntries := requestBody.Entries
+
+	if len(sendEntries) == 0 {
+		return utils.CreateErrorResponseV1("EmptyBatchRequest", true)
+	}
+
+	if len(sendEntries) > 10 {
+		return utils.CreateErrorResponseV1("TooManyEntriesInBatchRequest", true)
+	}
+	ids := map[string]struct{}{}
+	for _, v := range sendEntries {
+		if _, ok := ids[v.Id]; ok {
+			return utils.CreateErrorResponseV1("BatchEntryIdsNotDistinct", true)
+		}
+		ids[v.Id] = struct{}{}
+	}
+
+	sentEntries := make([]models.SendMessageBatchResultEntry, 0)
+	log.Debug("Putting Message in Queue:", queueName)
+	for _, sendEntry := range sendEntries {
+		msg := app.Message{MessageBody: []byte(sendEntry.MessageBody)}
+		if len(sendEntry.MessageAttributes) > 0 {
+			oldStyleMessageAttributes := convertToOldMessageAttributeValueStructure(sendEntry.MessageAttributes)
```

Comment:
I'm using the conversion step as SendMessageV1
https://github.com/Admiral-Piett/goaws/pull/288/files#diff-2b6e3902e73e67c2bf76a6c2ee3f3cc4be55d19105a543cf7954186e99c1ef54R63